src/libraries/System.Net.Quic/src/System/Net/Quic/Implementations/MsQuic/MsQuicStream.cs

```diff
@@ -257,12 +258,7 @@ internal override long StreamId
             get
             {
                 ThrowIfDisposed();
-
-                if (_streamId == -1)
-                {
-                    _streamId = GetStreamId();
-                }
-
+                Debug.Assert(_streamId != -1);
```

Comment:
I assume that in both directions (incoming/outgoing), the user cannot get a QuicStream instance before `_streamId` is set, correct?

Reply:
Correct.